journal/journal.go

```diff
@@ -0,0 +1,51 @@
+package journal
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// Journal defines an interface for recording operations and their metadata
+type Journal interface {
+	// Record records an operation and its metadata to a Journal accepting variadic key-value
+	// pairs.
+	Record(operation string, meta ...interface{})
+}
+
+// Builder defines a method for creating Journals with a topic.
+type Builder func(topic string) Journal
+
+// NewZapJournalBuilder returns a Builder backed by a zap logger.
+func NewZapJournalBuilder(filepath string) (Builder, error) {
```

Comment:
I think the topic string should be provided from the outside rather than determined within a component. So some naming things:
- NewZapJournalBuilder -> New
- JournalBuilder -> Journal
- Journal -> Writer
- Record() -> Write()

Rather than a func, use an interface for the ~Builder~Journal.

In the node, we'll then have:
```
j := journal.New(path)
syncer := chain.NewSyncer(..., j.Topic("chain"))
```
and in a component
```
  self.journal.Write("event", "key", value")
```